### PR TITLE
DOC: simplify CheckButton example

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Now that dictionaries are ordered, do not simulate an ordered dictionary via two lists.

### CODE REVIEW

File: examples/widgets/check_buttons.py [hunk 1]
```diff
@@ -25,21 +25,23 @@
...
+)
 
 
-def func(label):
-    index = labels.index(label)
-    lines[index].set_visible(not lines[index].get_visible())
+def callback(label):
+    ln = lines[label]
+    ln.set_visible(not ln.get_visible())
     plt.draw()
```
Comment (anntzer) [inline examples/widgets/check_buttons.py:42]: fig.canvas.draw_idle()?

File: examples/widgets/check_buttons.py [hunk 2]
```diff
@@ -25,21 +25,23 @@
...
 
 # Make checkbuttons with all plotted lines with correct visibility
 rax = fig.add_axes([0.05, 0.4, 0.1, 0.15])
-labels = [str(line.get_label()) for line in lines]
-visibility = [line.get_visible() for line in lines]
-check = CheckButtons(rax, labels, visibility)
+check = CheckButtons(
+    ax=rax,
+    labels=list(lines),
+    actives=[l.get_visible() for l in lines.values()]
```
Comment (timhoffm) [inline examples/widgets/check_buttons.py:35]: I don't think making `lines` a dict makes this necessarily better. It adds the cognitive load to rember what keys and values are (which is not not clear from the name `lines`.

I would leave `lines = [l0, l1, l2]`, which is very simple, and instead inline the comprehensions:

```suggestion
    labels=[l.get_label() for l in lines]
    actives=[l.get_visible() for l in lines]
```
This is also nicely symmetric in code structure.

File: examples/widgets/check_buttons.py [hunk 3]
```diff
@@ -25,21 +25,23 @@
...
+)
 
 
-def func(label):
-    index = labels.index(label)
-    lines[index].set_visible(not lines[index].get_visible())
+def callback(label):
+    ln = lines[label]
+    ln.set_visible(not ln.get_visible())
```
Comment (timhoffm) [inline examples/widgets/check_buttons.py:41]:
```suggestion
    line = next(l for l in lines if l.get_label() == label)
    line.set_visible(not line.get_visible())
```

**Comment by tacaswell** [comment]
@timhoffm I went with a better name rather than going back to a list.

**Comment by timhoffm** [comment]
I still like my suggestion better 🤷 but not it's not worth a discussion. I'll withdraw from here and let somebody else approve.

File: examples/widgets/check_buttons.py (hunk 3 above)
Comment (QuLogic) [reply examples/widgets/check_buttons.py]: I don't think using `next` is any less confusing than a dictionary, perhaps even more so. The alternative is using a full `for` loop, and at that point a dictionary lookup just seems less work.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**